### PR TITLE
Test PHPUnit in Travis in parallel

### PR DESCRIPTION
**Pull request by simison**
_Testing if it would make sense to run PHPUnit with different WP branches/modes combinations in parallel instead of running them in series._

_Also changed the way we install PHPUnit on older versions of PHP._

### Before
Average time to complete: 10m 16s 
image

### After
Average time to complete: 5m 50s
image

#### Testing instructions:
- TravisCI should spawn 20 parallel jobs
- Unit tests will run on PHP 5.2, 5.6, 7.0 and 7.2
- Lint and misc other tests will run only on PHP 5.6

### CODE REVIEW

**Comment by simison** [comment]
More discussion p9dueE-iz-p2

File: .travis.yml [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
-# Tell Travis CI which distro to use
-dist: trusty
+# Run Matrix for these PHP versions
+php:
+- "5.6"
+- "7.0"
+- "7.2"
+#- "7.0"
+#  env: "SWITCH_TO_PHP=5.2"
+#  dist: precise
```
Comment (simison) [inline .travis.yml]: Noting that these should probably be enabled at some point again as they're part of the initial setup:
```
#- "7.0"
#  env: "SWITCH_TO_PHP=5.2"
#  dist: precise
```

...I'm just not sure what exactly happens there. 🙃

Comment (abidhahmed) [reply .travis.yml]: Yep, gotcha. Looking into that now.

**Comment by abidhahmed** [comment]
Checks are failing due to instability on Hulu specific tests - not because of this PR.

**Comment by abidhahmed** [comment]
Adding back PHP 5.2 tests seem to decrease performance as the overall build time increases to be on par with what we have right now. I'll fiddle with build stages to see if it helps on #9838.

**Review by abidhahmed — COMMENTED** [review]
Works for me, but the time spent depends on how much TravisCI is loaded and how many parallel builds they'd allow at the same time.

**Comment by simison** [comment]
>Adding back PHP 5.2 tests seem to decrease performance as the overall build time increases to be on par with what we have right now. I'll fiddle with build stages to see if it helps on #9838.

Installing PHP 5.2 and related stuff seems to take over 5 minutes.

Do you know why we do:
```
- "7.0"
  env: "SWITCH_TO_PHP=5.2"
  dist: precise

```

Instead of directly:
```
- "5.2"
  dist: precise
```

More info about Travis and PHP 5.2: https://docs.travis-ci.com/user/reference/trusty#PHP-images

**Comment by abidhahmed** [comment]
@simison PHPUnit is unhappy when I did native `5.2` instead of running `7.0` without any additional setup:
https://travis-ci.org/Automattic/jetpack/jobs/399463824

Using `SWITCH_TO_PHP=5.2` on native `5.2` causes issue too:
https://travis-ci.org/Automattic/jetpack/jobs/399465436
https://travis-ci.org/Automattic/jetpack/builds/399465431

Seems we do this to use an older version of PHPUnit.

**Comment by simison** [comment]
>PHPUnit is unhappy when I did native `5.2` instead of running `7.0` without any additional setup:

PHPUnit that comes bundled with PHP 5.2, that is? Could we still use 5.2 from Travis and install only phpunit to one locations mentioned in this list? https://docs.travis-ci.com/user/languages/php#Default-Build-Script

Is `SWITCH_TO_PHP` a Travis thing or something custom to our build?

**Review by abidhahmed — COMMENTED** [review]
(double post, oops!)

**Comment by abidhahmed** [comment]
After fixing Hulu tests ~ test timings improved a LOT. However, I think that's sort of a gimmick, and is most likely TravisCI's nodes are a lot faster today with less loads compared to yesterday.
https://travis-ci.org/Automattic/jetpack/builds/399471512

**Comment by abidhahmed** [comment]
Update : It's generally faster from the look of things. I'll investigate more into PHP 5.2 and our custom setup.

Retested with a push to see if TravisCI cache interfered earlier:
https://travis-ci.org/Automattic/jetpack/builds/399970794

**Comment by simison** [comment]
image

⏱ 😍

**Comment by abidhahmed** [comment]
I was just testing #9864 ~ apart from the linting issues that get ignored by PHP 5.2, the builds themselves are flaky. I had to restart multiple times just to get it running.

This one in particular now is completely not working:
https://travis-ci.org/Automattic/jetpack/jobs/400257857

**Comment by simison** [comment]
Thanks for working on this @abidhahmed !

It's looking great.

Do you have thoughts on how this might perform at busy times in Travis? Could startup time be slowed down more than 5+mins?

If after merging we find out that this doesn't perform very well and we'd have to revert, are there any parts in this PR we'd like to keep anyway even when reverting parallelism? If so, those parts could be merged in a separate PR first.

File: .travis.yml [hunk 2]
```diff
@@ -1,45 +1,56 @@
...
-# Test WP trunk/master and two latest versions on minimum (5.2).
-# Test WP latest two versions (4.5, 4.3) on most popular (5.6, 7.0).
-# Test WP trunk/master on edge platforms (PHP 7.2).
-
-# WP_VERSION specifies the tag to use. The way these tests are configured to run
-# requires at least WordPress 3.8. Specify "master" to test against SVN trunk.
-
 matrix:
   include:
+  # Versions listed below run within Precise distribution with PHP 5.2
```
Comment (simison) [inline .travis.yml]: Perhaps we could elaborate here that only `precise` (and not `trusty`) dist has PHP 5.2 available?

I.e. comment on _why_, not only _what_

Comment (abidhahmed) [reply .travis.yml]: I've updated the comment to include this, thanks!

**Comment by abidhahmed** [comment]
This is ready for prime time, latest tests are so fast I can't stop smiling.
https://travis-ci.org/Automattic/jetpack/jobs/402534754

**Comment by simison** [comment]
Looks brilliant and whoa it's fast! ⏱⚡️✨

Can't press _"accept"_ on PR I started but thumbs up from me.

**Review by brbrr — APPROVED** [review]
That's a great work @abidhahmed & @simison!

:shipit:

**Pull request merged**